handlers.go

```diff
@@ -4208,11 +4209,40 @@ func (s *Server) handleColfsListDir(c echo.Context, u *User) error {
 			return err
 		}
 
+		// Build a new local DAGService to query for the CID if our ref and see if it's a directory
+		bserv := blockservice.New(s.Node.Blockstore, offline.Exchange(s.Node.Blockstore))
+		dserv := merkledag.NewDAGService(bserv)
+		ctx := c.Request().Context()
+		nd, err := dserv.Get(ctx, r.Cid.CID)
+		if err != nil {
+			return err
+		}
+
 		// if the relative path requires pathing up, its definitely not in this dir
 		if strings.HasPrefix(relp, "..") {
 			continue
 		}
 
+		// TODO: maybe find a way to reuse s.Node or s.gwayHandler.dserv
+		// Need to transform node into a unixfs node to check if it's dir
+		fsNode, err := unixfs.ExtractFSNode(nd)
+		if err != nil {
+			return err
```

Comment:
similar to my above comment -- I assume it might be ok that extracting an FS node from a CID is allowed to fail for raw nodes, and that shouldn't (necessarily) fail to list the whole directory. in that case, it seems like listing it as a file would be just fine

Reply:
Hummm I don't know if I'm too picky with myself, but listing it as a file just doesn't seem good enough. I like the point about not failing all the query though. Maybe we could change the `File: true/false` to a field like `Type: file/dir/undefined`?

Reply:
So I changed it to this: if it can be cast to a unixfs node then we can determine whether its `CidType` is a `File` or a `Dir`, otherwise it's set to `Raw`